test/common/router/config_impl_test.cc

```diff
@@ -3776,241 +3591,181 @@ TEST(NullConfigImplTest, All) {
-      "routes must specify one of prefix/path/regex");
-  ;
+  EXPECT_THROW(TestConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), factory_context_, true),
+               EnvoyException);
 }
 
-TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigPrefixAndPathAndRegex) {
-  const std::string json = R"EOF(
-  {
-    "virtual_hosts": [
-      {
-        "name": "www2",
-        "domains": ["*"],
-        "routes": [
-          {
-            "prefix": "/",
-            "path": "/foo",
-            "regex": "/[bc]at",
-            "cluster": "www2"
-          }
-        ]
-      }
-    ]
-  }
+TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigNoAction) {
+  const std::string yaml = R"EOF(
+virtual_hosts:
+- name: www2
+  domains:
+  - "*"
+  routes:
+  - match:
+      prefix: "/api"
   )EOF";
-
-  EXPECT_THROW_WITH_MESSAGE(
-      TestConfigImpl(parseRouteConfigurationFromJson(json), factory_context_, true), EnvoyException,
-      "routes must specify one of prefix/path/regex");
+  EXPECT_THROW(TestConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), factory_context_, true),
+               EnvoyException);
 }
 
-TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigMissingPathSpecifier) {
-  const std::string json = R"EOF(
-  {
-    "virtual_hosts": [
-      {
-        "name": "www2",
-        "domains": ["*"],
-        "routes": [
-          {
-            "cluster": "www2"
-          }
-        ]
-      }
-    ]
-  }
+TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigPathAndRegex) {
+  const std::string yaml = R"EOF(
+virtual_hosts:
+- name: www2
+  domains:
+ - "*"
+  routes:
+  - match:
+      path: "/foo"
+      regex: "/[bc]at"
+    route:
+      cluster: www2
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(
-      TestConfigImpl(parseRouteConfigurationFromJson(json), factory_context_, true), EnvoyException,
-      "routes must specify one of prefix/path/regex");
+  EXPECT_THROW(TestConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), factory_context_, true),
+               EnvoyException);
 }
 
-TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigNoRedirectNoClusters) {
-  const std::string json = R"EOF(
-  {
-    "virtual_hosts": [
-      {
-        "name": "www2",
-        "domains": ["*"],
-        "routes": [
-          {
-            "prefix": "/api"
-          }
-        ]
-      }
-    ]
-  }
+TEST_F(BadHttpRouteConfigurationsTest, BadRouteEntryConfigPrefixAndPathAndRegex) {
+  const std::string yaml = R"EOF(
+virtual_hosts:
+- name: www2
+  domains:
+  - "*"
+  routes:
+  - match:
+      prefix: "/"
+      path: "/foo"
+      regex: "/[bc]at"
+    route:
+      cluster: www2
   )EOF";
 
-  EXPECT_THROW_WITH_MESSAGE(
-      TestConfigImpl(parseRouteConfigurationFromJson(json), factory_context_, true), EnvoyException,
-      "routes must have redirect or one of cluster/cluster_header/weighted_clusters")
+  EXPECT_THROW(TestConfigImpl(parseRouteConfigurationFromV2Yaml(yaml), factory_context_, true),
```

Comment:
Did this change because it's now handled by validation? Is it possible to include the validation error message? Same above?

Reply:
Ah yeah, forgot to come back to these after figuring out protobuf validation. Handled now. Using `_WITH_REGEX` with the critical message part since the protobuf errors are pretty verbose